File: frontend/src/types/review.ts
export interface ReviewContainer {
  id: string;
  name: string;
  type: string | null;
  location_description: string | null;
}

export interface ReviewImageAsset {
  image_asset_id: string;
  original_filename: string | null;
  stored_filename: string | null;
  file_path: string;
  thumbnail_path: string | null;
  mime_type: string | null;
  file_size_bytes: number | null;
  status: string;
  upload_order: number;
}

export interface ReviewUploadGroup {
  upload_group_id: string;
  upload_session_id: string;
  container: ReviewContainer | null;
  inventory_group_id: string | null;
  inventory_group_code: string | null;
  inventory_group_name: string | null;
  client_group_id: string | null;
  title: string | null;
  notes: string | null;
  sort_order: number;
  status: string;
  created_datetime: string;
  updated_datetime: string | null;
  image_count: number;
  processed_image_count: number;
  failed_image_count: number;
  ai_assist_status: 'not_requested' | 'queued' | 'processing' | 'succeeded' | 'failed';
  ai_assist_error_message: string | null;
  ai_assist_requested_datetime: string | null;
  ai_assist_started_datetime: string | null;
  ai_assist_completed_datetime: string | null;
  ai_suggested_title: string | null;
  ai_suggested_description: string | null;
  ai_suggested_approx_value: string | null;
  images: ReviewImageAsset[];
}

export interface ReviewQueueResponse {
  groups: ReviewUploadGroup[];
}

export interface GetReviewUploadGroupResponse {
  group: ReviewUploadGroup;
}

export interface ReviewUploadGroupImageMutationResponse {
  group: ReviewUploadGroup;
}

export interface ApproveUploadGroupInput {
  title?: string;
  description?: string;
  approx_value?: string;
  sold_date?: string | null;
  notes?: string | null;
  inventory_group_id?: string;
}

export interface ReviewItem {
  id: string;
  container_id: string | null;
  inventory_group_id: string | null;
  inventory_group_code: string | null;
  inventory_group_name: string | null;
  title: string | null;
  description: string | null;
  approx_value: string | null;
  sold_date: string | null;
  notes: string;
  ai_enriched: boolean;
  created_datetime: string;
  updated_datetime: string | null;
}

export interface ApproveUploadGroupResponse {
  item: ReviewItem;
  linked_image_count: number;
}

export interface QueueReviewAIAssistResponse {
  upload_group_id: string;
  ai_assist_status: 'queued' | 'processing' | 'succeeded' | 'failed' | 'not_requested';
  ai_assist_error_message: string | null;
  ai_suggested_title: string | null;
  ai_suggested_description: string | null;
  ai_suggested_approx_value: string | null;
  ai_assist_requested_datetime: string | null;
  ai_assist_started_datetime: string | null;
  ai_assist_completed_datetime: string | null;
}

export interface QueueReviewAIAssistInput {
  user_hint?: string;
}

export interface ReviewGroupDeletePreview {
  upload_group_id: string;
  upload_session_id: string;
  client_group_id: string | null;
  title: string | null;
  image_count: number;
  file_count: number;
  total_file_size_bytes: number;
  warnings: string[];
  files: Array<{
    image_asset_id: string;
    kind: string;
    path: string;
    exists: boolean;
    size_bytes: number;
  }>;
}

export interface ReviewGroupDeleteResponse {
  deleted_upload_group_id: string;
  deleted_image_asset_count: number;
  deleted_file_count: number;
  missing_file_count: number;
  warnings: string[];
}

export interface ReviewImageDeleteResponse {
  group: ReviewUploadGroup | null;
  deleted_image_asset_id: string;
  deleted_file_count: number;
  missing_file_count: number;
  warnings: string[];
}

File: api/internal/models/review.go
package models

import "time"

type ReviewContainer struct {
	ID                  string  `json:"id"`
	Name                string  `json:"name"`
	Type                *string `json:"type"`
	LocationDescription *string `json:"location_description"`
}

type ReviewImageAsset struct {
	ImageAssetID     string  `json:"image_asset_id"`
	OriginalFilename *string `json:"original_filename"`
	StoredFilename   *string `json:"stored_filename"`
	FilePath         string  `json:"file_path"`
	ThumbnailPath    *string `json:"thumbnail_path"`
	NormalizedPath   *string `json:"normalized_path"`
	MimeType         *string `json:"mime_type"`
	FileSizeBytes    *int64  `json:"file_size_bytes"`
	Status           string  `json:"status"`
	UploadOrder      int     `json:"upload_order"`
}

type ReviewUploadGroup struct {
	UploadGroupID             string             `json:"upload_group_id"`
	UploadSessionID           string             `json:"upload_session_id"`
	Container                 *ReviewContainer   `json:"container"`
	LocationID                *string            `json:"location_id"`
	LocationName              *string            `json:"location_name"`
	LocationDetail            *string            `json:"location_detail"`
	InventoryGroupID          *string            `json:"inventory_group_id"`
	InventoryGroupCode        *string            `json:"inventory_group_code"`
	InventoryGroupName        *string            `json:"inventory_group_name"`
	ClientGroupID             *string            `json:"client_group_id"`
	Title                     *string            `json:"title"`
	Notes                     *string            `json:"notes"`
	SortOrder                 int                `json:"sort_order"`
	Status                    string             `json:"status"`
	CreatedDatetime           time.Time          `json:"created_datetime"`
	UpdatedDatetime           *time.Time         `json:"updated_datetime"`
	ImageCount                int                `json:"image_count"`
	ProcessedImageCount       int                `json:"processed_image_count"`
	FailedImageCount          int                `json:"failed_image_count"`
	AIAssistStatus            string             `json:"ai_assist_status"`
	AIAssistErrorMessage      *string            `json:"ai_assist_error_message"`
	AIAssistRequestedDatetime *time.Time         `json:"ai_assist_requested_datetime"`
	AIAssistStartedDatetime   *time.Time         `json:"ai_assist_started_datetime"`
	AIAssistCompletedDatetime *time.Time         `json:"ai_assist_completed_datetime"`
	AISuggestedTitle          *string            `json:"ai_suggested_title"`
	AISuggestedDescription    *string            `json:"ai_suggested_description"`
	AISuggestedApproxValue    *string            `json:"ai_suggested_approx_value"`
	Images                    []ReviewImageAsset `json:"images"`
}

type ReviewUploadGroupList struct {
	Groups []ReviewUploadGroup `json:"groups"`
}

type GetReviewUploadGroupResponse struct {
	Group ReviewUploadGroup `json:"group"`
}

type ReviewUploadGroupImageMutationResponse struct {
	Group ReviewUploadGroup `json:"group"`
}

type ApproveUploadGroupRequest struct {
	Title            *string `json:"title"`
	Description      *string `json:"description"`
	ApproxValue      *string `json:"approx_value"`
	SoldDate         *string `json:"sold_date"`
	Notes            *string `json:"notes"`
	InventoryGroupID *string `json:"inventory_group_id"`
}

type ReviewItem struct {
	ID                 string     `json:"id"`
	ContainerID        *string    `json:"container_id"`
	LocationID         *string    `json:"location_id"`
	LocationName       *string    `json:"location_name"`
	LocationDetail     *string    `json:"location_detail"`
	InventoryGroupID   *string    `json:"inventory_group_id"`
	InventoryGroupCode *string    `json:"inventory_group_code"`
	InventoryGroupName *string    `json:"inventory_group_name"`
	Title              *string    `json:"title"`
	Description        *string    `json:"description"`
	ApproxValue        *string    `json:"approx_value"`
	SoldDate           *string    `json:"sold_date"`
	Notes              string     `json:"notes"`
	AiEnriched         bool       `json:"ai_enriched"`
	CreatedDatetime    time.Time  `json:"created_datetime"`
	UpdatedDatetime    *time.Time `json:"updated_datetime"`
}

type ApproveUploadGroupResponse struct {
	Item             ReviewItem `json:"item"`
	LinkedImageCount int        `json:"linked_image_count"`
}

type QueueReviewAIAssistResponse struct {
	UploadGroupID             string     `json:"upload_group_id"`
	AIAssistStatus            string     `json:"ai_assist_status"`
	AIAssistErrorMessage      *string    `json:"ai_assist_error_message"`
	AISuggestedTitle          *string    `json:"ai_suggested_title"`
	AISuggestedDescription    *string    `json:"ai_suggested_description"`
	AISuggestedApproxValue    *string    `json:"ai_suggested_approx_value"`
	AIAssistRequestedDatetime *time.Time `json:"ai_assist_requested_datetime"`
	AIAssistStartedDatetime   *time.Time `json:"ai_assist_started_datetime"`
	AIAssistCompletedDatetime *time.Time `json:"ai_assist_completed_datetime"`
}

type QueueReviewAIAssistRequest struct {
	UserHint *string `json:"user_hint"`
}

type ReviewUploadGroupDeletePreview struct {
	UploadGroupID      string              `json:"upload_group_id"`
	UploadSessionID    string              `json:"upload_session_id"`
	ClientGroupID      *string             `json:"client_group_id"`
	Title              *string             `json:"title"`
	ImageCount         int                 `json:"image_count"`
	FileCount          int                 `json:"file_count"`
	TotalFileSizeBytes int64               `json:"total_file_size_bytes"`
	Warnings           []string            `json:"warnings"`
	Files              []DeletePreviewFile `json:"files"`
}

type ReviewUploadGroupDeleteResponse struct {
	DeletedUploadGroupID   string   `json:"deleted_upload_group_id"`
	DeletedImageAssetCount int      `json:"deleted_image_asset_count"`
	DeletedFileCount       int      `json:"deleted_file_count"`
	MissingFileCount       int      `json:"missing_file_count"`
	Warnings               []string `json:"warnings"`
}

type ReviewImageDeleteResponse struct {
	Group               *ReviewUploadGroup `json:"group"`
	DeletedImageAssetID string             `json:"deleted_image_asset_id"`
	DeletedFileCount    int                `json:"deleted_file_count"`
	MissingFileCount    int                `json:"missing_file_count"`
	Warnings            []string           `json:"warnings"`
}
